pkg/api/v1/types.go

```diff
@@ -2064,6 +2064,32 @@ type ReplicationControllerStatus struct {
 
 	// ObservedGeneration reflects the generation of the most recently observed replication controller.
 	ObservedGeneration int64 `json:"observedGeneration,omitempty" protobuf:"varint,3,opt,name=observedGeneration"`
+
+	// Represents the latest available observations of a replication controller's current state.
+	Conditions []ReplicationControllerCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+}
+
+type ReplicationControllerConditionType string
+
+// These are valid conditions of a replication controller.
+const (
+	// ReplicationControllerReplicaFailure is added in a replication controller when one of its pods
+	// fails to be created or deleted.
+	ReplicationControllerReplicaFailure ReplicationControllerConditionType = "ReplicaFailure"
+)
+
+// ReplicationControllerCondition describes the state of a replication controller at a certain point.
+type ReplicationControllerCondition struct {
+	// Type of replication controller condition.
+	Type ReplicationControllerConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status ConditionStatus `json:"status"`
+	// The last time the condition transitioned from one status to another.
+	LastTransitionTime unversioned.Time `json:"lastTransitionTime,omitempty"`
```

Comment:
Usually, there's also `LastProbeTime unversioned.Time` which is the last time a check was actually performed. See [PodCondition](https://github.com/kubernetes/kubernetes/blob/42e5f95a6b41877a56f909e06aef912f63948eaa/pkg/api/types.go#L1233) or [JobCondition](https://github.com/kubernetes/kubernetes/blob/42e5f95a6b41877a56f909e06aef912f63948eaa/pkg/apis/batch/types.go#L159).